test/src/node.rs

```diff
@@ -94,6 +96,43 @@ impl Node {
         &self.working_dir
     }
 
+    pub fn log_file(&self) -> String {
+        node_log(&self.working_dir()).display().to_string()
+    }
+
+    pub fn log_monitor(&self, expected: &str, timeout: u64) -> Receiver<String> {
+        let (log_sender, log_notifier) = bounded(1);
+        let timeout = tweaked_duration(timeout);
+        let start = Instant::now();
+        let filename = &self.log_file();
+        let mut skip_lines = 0;
+        let mut modified = SystemTime::now();
+        loop {
+            let filename = fs::File::open(&filename).unwrap();
```

Comment:
It's faster to keep the file handle, and poll the new content.

I found several crates that you can refer to:

* https://crates.io/crates/chase async file following
* https://crates.io/crates/logwatcher tail -F clone
* https://crates.io/crates/logwatch another tail -F